runtime/altair/src/migrations.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_types::tokens::CurrencyId;
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+
+use crate::{OrmlAssetRegistry, Runtime};
+
+pub type UpgradeAltair1027 = (
+	pallet_loans_ref::migrations::v1::Migration<Runtime>,
+	TrancheLocationMigration,
+);
+
+/// This migration sets the AssetMetadata.location of all the Tranche tokens
+/// registered in the AssetRegistry to `None`.
+pub struct TrancheLocationMigration;
+
+impl OnRuntimeUpgrade for TrancheLocationMigration {
+	fn on_runtime_upgrade() -> Weight {
+		for (asset_id, metadata) in orml_asset_registry::Metadata::<Runtime>::iter() {
+			if matches!(asset_id, CurrencyId::Tranche(_, _)) && metadata.location.is_some() {
+				match OrmlAssetRegistry::do_update_asset(
+					asset_id,
+					// decimals
+					None,
+					// name
+					None,
+					// symbol
+					None,
+					// existential_deposit
+					None,
+					// location: we do set it to `None`
+					Some(None),
+					// additional
+					None,
+				) {
+					Err(e) => log::error!("TrancheLocationMigration: Failed to update asset with underlying error: {:?}", e),
+					_ => continue,
```

Comment:
Nit: Since we are matching for two possible outcomes and only working on the error variant, it would be cleaner to map the error and then drop the result instead.

```suggestion
				OrmlAssetRegistry::do_update_asset(
					asset_id,
					// decimals
					None,
					// name
					None,
					// symbol
					None,
					// existential_deposit
					None,
					// location: we do set it to `None`
					Some(None),
					// additional
					None,
				)
				.map_err(|e| {
					log::error!("TrancheLocationMigration: Failed to update asset with underlying error: {:?}", e);
				})
				.ok();
```

Reply:
Applied :) I forgot that in Rust a map can be just a closure that doesn't return a value per se. Thanks!